src/variable.jl

```diff
@@ -351,19 +350,20 @@ function _write_data_to_nc(v::Variable{T,N},data,indexes::Int...) where {T,N}
     nc_put_var1(v.ds.ncid,v.varid,[i-1 for i in indexes[ndims(v):-1:1]],T(data[1]))
 end
 
-_write_data_to_nc(v::Variable,data) = _write_data_to_nc(v, data, 1)
+_write_data_to_nc(v::Variable, data) = _write_data_to_nc(v, data, 1)
 
-function _write_data_to_nc(v::Variable,data,indexes::Colon...)
+function _write_data_to_nc(v::Variable, data, indexes::Colon...)
+    @info "Am I even getting here ?"
     nc_put_var(v.ds.ncid,v.varid,data)
 end
 
-function _write_data_to_nc(v::Variable,data,indexes::StepRange{Int,Int}...)
+function _write_data_to_nc(v::Variable{T, N}, data, indexes::StepRange{Int,Int}...) where {T, N}
     start,count,stride,jlshape = ncsub(indexes[1:ndims(v)])
-    nc_put_vars(v.ds.ncid,v.varid,start,count,stride,data)
+    nc_put_vars(v.ds.ncid,v.varid,start,count,stride,T.(data))
```

Comment:
Probably DiskArrays.jl wont let you do this optimisation either. You may need to override these methods on `OrdinalRange` https://github.com/meggart/DiskArrays.jl/blob/main/src/batchgetindex.jl#L166-L216 so that can happen here rather than in DiskArrays.jl

Reply:
I think by defining `readblock!` and `writeblock!` for StepRanges should be sufficient to go into these optimisations. DiskArrays does provide some fallback routines in case only `AbstractUnitRange`s are supported by the backend but they should not be called in this case.